### PR TITLE
docs(): Update Federation Docs

### PR DESCRIPTION
**Pull request by zgriesinger**
Updating the docs to include clearer info on how to create the necessary set up in the base type/service with `resolveReference` and `GraphQLFederationModule`.

### CODE REVIEW

File: documentation/docs/graphql/federation.mdx [hunk 1]
```diff
@@ -24,7 +24,65 @@ The `__typename` lets the gateway know which type is being referenced with addit
 
 ### References in Resolvers
 
-To reference a type in `nestjs-query` you must first create a type that extends the base type contained in some other graphql service.
+To reference a type in `nestjs-query` you must first create a federated service that defines the base type.
+
+#### Base Type
+The base type must include be decorated with federated directives specifying key
```
Comment (doug-martin) [inline documentation/docs/graphql/federation.mdx:30]: Fix this sentence :)

Comment (doug-martin) [reply documentation/docs/graphql/federation.mdx:30]: We may also want to call out that this is in another service from the rest of the example below.

File: documentation/docs/graphql/federation.mdx [hunk 2]
```diff
@@ -212,6 +270,8 @@ In the above example we extend `FederationResolver` this will not set up any que
 
 The final step is to register your module just like normal
 
+:::note Make sure you are importing the `GraphQLFederatedModule` in the root of every service included in your federated graph.
+
```
Comment (doug-martin) [inline documentation/docs/graphql/federation.mdx:274]: this should be 
```
:::note
Make sure you are importing the `GraphQLFederatedModule` in the root of every service included in your federated graph.
:::
```

This is syntax used by docusaurus to create callouts. https://v2.docusaurus.io/docs/markdown-features#calloutsadmonitions

File: documentation/docs/graphql/federation.mdx (hunk 1 above)
Comment (zgriesinger) [reply documentation/docs/graphql/federation.mdx]: Made this sentence make sense and added a bit before the extending type section making it clear that it is two separate services.

**Pull request merged**